src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/tasks/OrchestratorNotificationTask.java

```diff
@@ -46,19 +48,42 @@ public OrchestratorNotificationTask(
     @SchedulerLock(name = "send-orchestrator-notification")
     @Scheduled(fixedDelayString = "${scheduling.task.notifications_to_orchestrator.delay}")
     public void run() {
-        envelopeRepo
-            .findByStatus(PROCESSED)
-            .forEach(env -> {
+        log.info("Started sending notifications to orchestrator");
+
+        List<Envelope> envelopesToSend = envelopeRepo.findByStatus(PROCESSED);
+
+        int successCount = (int)envelopesToSend
+            .stream()
+            .map(env -> {
                 try {
                     serviceBusHelper.sendMessage(new EnvelopeMsg(env));
+                    logEnvelopeSent(env);
                     createEvent(env, Event.DOC_PROCESSED_NOTIFICATION_SENT);
                     updateStatus(env);
+                    return 1;
                 } catch (Exception exc) {
                     createEvent(env, Event.DOC_PROCESSED_NOTIFICATION_FAILURE);
                     // log error and try with another envelope.
-                    LOGGER.error("Error sending envelope notification", exc);
+                    log.error("Error sending envelope notification", exc);
+                    return 0;
                 }
-            });
+            })
+            .count();
```

Comment:
i think `.count` will return the size of the stream. instead we should use `.mapToInt` and `.sum` directly to actually reduce properly and get the correct count of success

Reply:
Yes, of course. Or I could filter instead of mapping. Well spotted.